FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/FhirToDataLakeOrchestratorJob.cs

```diff
@@ -0,0 +1,428 @@
+        private FhirToDataLakeOrchestratorJobResult _result;
+
+        public FhirToDataLakeOrchestratorJob(
+            JobInfo jobInfo,
+            FhirToDataLakeOrchestratorJobInputData inputData,
+            FhirToDataLakeOrchestratorJobResult result,
+            IFhirDataClient dataClient,
+            IFhirDataWriter dataWriter,
+            IQueueClient queueClient,
+            ITypeFilterParser typeFilterParser,
+            IGroupMemberExtractor groupMemberExtractor,
+            TableClient metaDataTableClient,
+            JobSchedulerConfiguration schedulerConfiguration,
+            FilterConfiguration filterConfiguration,
+            ILogger<FhirToDataLakeOrchestratorJob> logger)
+        {
+            _jobInfo = EnsureArg.IsNotNull(jobInfo, nameof(jobInfo));
+            _inputData = EnsureArg.IsNotNull(inputData, nameof(inputData));
+            _result = EnsureArg.IsNotNull(result, nameof(result));
+            _dataClient = EnsureArg.IsNotNull(dataClient, nameof(dataClient));
+            _dataWriter = EnsureArg.IsNotNull(dataWriter, nameof(dataWriter));
+            _queueClient = EnsureArg.IsNotNull(queueClient, nameof(queueClient));
+            _typeFilterParser = EnsureArg.IsNotNull(typeFilterParser, nameof(typeFilterParser));
+            _groupMemberExtractor = EnsureArg.IsNotNull(groupMemberExtractor, nameof(groupMemberExtractor));
+            _metaDataTableClient = EnsureArg.IsNotNull(metaDataTableClient, nameof(metaDataTableClient));
+            _metaDataTableClient.CreateIfNotExists();
+            _schedulerConfiguration = EnsureArg.IsNotNull(schedulerConfiguration, nameof(schedulerConfiguration));
+            _filterConfiguration = EnsureArg.IsNotNull(filterConfiguration, nameof(filterConfiguration));
+
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+        }
+
+        public int InitialOrchestrationIntervalInSeconds { get; set; } = JobConfigurationConstants.DefaultInitialOrchestrationIntervalInSeconds;
+
+        public int IncrementalOrchestrationIntervalInSeconds { get; set; } = JobConfigurationConstants.DefaultIncrementalOrchestrationIntervalInSeconds;
+
+        public int CheckFrequencyInSeconds { get; set; } = JobConfigurationConstants.DefaultCheckFrequencyInSeconds;
+
+        public int NumberOfPatientsPerProcessingJob { get; set; } = JobConfigurationConstants.DefaultNumberOfPatientsPerProcessingJob;
+
+        public async Task<string> ExecuteAsync(IProgress<string> progress, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Start executing FhirToDataLake orchestrator job {_jobInfo.GroupId}");
+
+            try
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    throw new OperationCanceledException();
+                }
+
+                // for group scope, extract patient list from group at first
+                var toBeProcessedPatients = new List<PatientWrapper>();
```

Comment:
Looks like this is only for group filter? Then we should put it together with group case, and extract to separate method or class.

Reply:
Refine the implementation per your suggestion.